In [31]:
import pathlib
import pandas as pd
import xml.etree.ElementTree as et

# define the path
currentDirectory = pathlib.Path('XML')

# define the pattern
currentPattern = "*.xml"
all_xml_files = []
for currentFile in currentDirectory.glob(currentPattern):
    all_xml_files.append(currentFile.parts[-1])
    
print(all_xml_files)

['Brother_Bear.xml', 'American_Pie.xml', 'Andaz_Apna_Apna.xml', 'The_Fault_in_Our_Stars.xml', 'Shakespeare_in_Love.xml', 'Night_at_the_Museum__Secret_of_the_Tomb.xml', 'Live_Nude_Girls.xml', 'The_Babadook.xml', 'The_Hobbit__The_Desolation_of_Smaug.xml', 'Boogie_Nights.xml', 'V_for_Vendetta.xml', 'The_Theory_of_Everything.xml', 'Robin_Hood.xml', 'The_Phantom_Tollbooth.xml', 'Ode_to_My_Father.xml', 'Mulan.xml', 'Bhoothnath_Returns.xml', 'The_Reluctant_Dragon.xml', 'The_Dark_Knight.xml', 'Elsa___Fred.xml', 'Lord_of_the_Flies.xml', 'My_Week_with_Marilyn.xml', 'Mean_Girls.xml', 'Invincible.xml', 'Hotel_Transylvania.xml', 'Khumba.xml', 'The_Bourne_Legacy.xml', 'Bros_Before_Hos.xml', 'Hook.xml', 'Black_Swan.xml', 'Still_Alice.xml', 'Exodus__Gods_and_Kings.xml', 'Chef.xml', 'The_Hunger_Games__Catching_Fire.xml', 'Into_the_Wild.xml', 'A_Single_Man.xml', 'The_Blair_Witch_Project.xml', 'Looney_Tunes__Back_in_Action.xml', 'The_Reader.xml', 'Total_Recall.xml', 'The_Imitation_Game.xml', 'Maine_Pyar_

In [32]:
def parse_XML(xml_file, name, df_cols):
            global out_df

            xtree = et.parse(xml_file)
            xroot = xtree.getroot()
            fin = []
            res = []
            res.append(name.split(".")[0])
            
            for node in xroot:
                for i in node_cols:
                    res.append(node.attrib.get(i))

            fin.append(res)
            print(fin)
            out_df = pd.DataFrame(data=fin, columns =meta_cols)
            return out_df
        
def parse_XML1(xml_file, name, df_cols, df):
            global out_df
            xtree = et.parse(xml_file)
            xroot = xtree.getroot()
            fin = []
            res = []
            if name.split(".")[0] == "Mr":
                res.append("Mr._Go")
            else:
                res.append(name.split(".")[0])
            for node in xroot:
                for i in node_cols:
                    res.append(node.attrib.get(i))

            fin.append(res)
            df1 = pd.DataFrame(data=fin, columns =meta_cols)
            
            frames = [df, df1]
            
            out_df = pd.concat(frames)
            
            return out_df

In [33]:
meta_cols = ["filename","language", "year", "genre", "country",
            "runtime", "rated"]

node_cols = ["language", "year", "genre", "country",
            "runtime", "rated"]

user_rating_cols = ["metascore", "imdbRating", "tomatoRating", "tomatoUserRating"]
user_and_meta_cols = ["language", "year", "genre", "country",
            "runtime", "rated", "metascore", "imdbRating", "tomatoRating", "tomatoUserRating"]
user = True
meta = True

In [34]:
if user == True:
    #meta_cols=["filename","metascore", "imdbRating", "tomatoMeter", "tomatoUserMeter"]
    meta_cols=["filename","language", "year", "genre", "country",
            "runtime", "rated"]
    node_cols=user_rating_cols
if user == True and meta == True:
    meta_cols=["filename","language", "year", "genre", "country",
            "runtime", "rated", "metascore", "imdbRating", "tomatoRating", "tomatoUserRating"]
    node_cols=user_and_meta_cols
c = 1
for i in all_xml_files:
    if c==1:
        parse_XML("XML/" + i, i,  meta_cols)
        c=43
    else:
        parse_XML1("XML/"+i, i, meta_cols, out_df)

[['Brother_Bear', 'English, Inuktitut, Croatian, Serbian', '2003', 'Animation, Adventure, Family', 'USA', '85 min', 'G', '48', '6.7', '5.5', '3.3']]


In [35]:
df_ground = pd.read_excel("dev_set_groundtruth_and_trailers.xls",index_col=False)

In [36]:
a = df_ground["filename"].sort_values()
b = out_df["filename"].sort_values()

for i in a.values:
    if i in b.values:
        continue
    else:
        print(i)

In [37]:
df_join = pd.merge(df_ground, out_df, on="filename", how="inner")
#df_join = df_join.drop(["movie", "filename", "trailer"], axis=1)
df_join

movie                         filename  \
0                       Seventh Son                      Seventh_Son   
1                     Welcome to Me                    Welcome_to_Me   
2                         The Judge                        The_Judge   
3   Transformers  Age of Extinction  Transformers__Age_of_Extinction   
4                  The Normal Heart                 The_Normal_Heart   
..                              ...                              ...   
90                  Live Nude Girls                  Live_Nude_Girls   
91                     Pulp Fiction                     Pulp_Fiction   
92                            Perez                            Perez   
93                    A Goofy Movie                    A_Goofy_Movie   
94                       Mean Girls                       Mean_Girls   

    goodforairplane                                            trailer  \
0                 1  http://www.imdb.com/video/imdb/vi1803464473/?r...   
1                 0  http://www.imdb.com/video/imdb/vi226471449/?re...   
2                 0  http://www.imdb.com/video/imdb/vi1286123289/?r...   
3                 0  http://www.imdb.com/video/imdb/vi3138759961/?r...   
4                 1  http://www.imdb.com/video/imdb/vi2749541657/?r...   
..              ...                                                ...   
90                0        https://www.youtube.com/watch?v=RrjF-FSFrfo   
91                0        https://www.youtube.com/watch?v=s7EdQ4FqbhY   
92                0        https://www.youtube.com/watch?v=HsXRKgzs1Kk   
93                0        https://www.youtube.com/watch?v=Ou94SEYa9jI   
94                0  http://www.imdb.com/video/imdb/vi3868524825/?r...   

                                language  year                         genre  \
0                                English  2014    Action, Adventure, Fantasy   
1                                English  2014                 Comedy, Drama   
2                                English  2014                         Drama   
3                                English  2014     Action, Adventure, Sci-Fi   
4                                English  2014                         Drama   
..                                   ...   ...                           ...   
90                               English  1995                        Comedy   
91              English, Spanish, French  1994                  Crime, Drama   
92                               Italian  2014        Crime, Drama, Thriller   
93                               English  1995  Animation, Adventure, Comedy   
94  English, German, Vietnamese, Swahili  2004                        Comedy   

                   country  runtime  rated metascore imdbRating tomatoRating  \
0   USA, UK, Canada, China  102 min  PG-13        30        5.5          3.8   
1                      USA  105 min      R        67        6.2          6.4   
2                      USA  141 min      R        48        7.5          5.6   
3               USA, China  165 min  PG-13        32        5.8          3.9   
4                      USA  132 min  TV-MA       N/A        8.0          7.7   
..                     ...      ...    ...       ...        ...          ...   
90                     USA   95 min      R       N/A        5.4          4.4   
91                     USA  154 min      R        94        8.9          9.0   
92                   Italy   94 min    N/A       N/A        6.5          N/A   
93                     USA   78 min      G       N/A        6.8          5.6   
94             USA, Canada   97 min  PG-13        66        7.0          6.9   

   tomatoUserRating  
0               2.9  
1               3.4  
2               3.8  
3               3.3  
4               4.2  
..              ...  
90              3.2  
91              4.2  
92              4.2  
93              3.1  
94              3.0  

[95 rows x 14 columns]

In [38]:
import numpy as np
if user == False or meta == True:
    count=0
    for i in df_join["genre"].values:
        df_join["genre"].values[count] = i.replace(" ", "")
        count+=1

    count=0
    for i in df_join["language"].values:
        df_join["language"].values[count] = i.replace(" ", "")
        count+=1


    count=0
    for i in df_join["country"].values:
        df_join["country"].values[count] = i.replace(" ", "")
        count+=1
        
    genre = df_join.genre.str.split(",", expand=True).stack()
    language = df_join.language.str.split(",", expand=True).stack()
    country = df_join.country.str.split(",", expand=True).stack()
    
    df_join = pd.concat([df_join, 
                pd.get_dummies(genre, prefix="g").groupby(level=0).sum()],axis=1) \
                .drop(["genre"],axis=1)

    df_join = pd.concat([df_join, 
                    pd.get_dummies(language, prefix="l").groupby(level=0).sum()],axis=1) \
                    .drop(["language"],axis=1)

    df_join = pd.concat([df_join, 
                    pd.get_dummies(country, prefix="c").groupby(level=0).sum()],axis=1) \
                    .drop(["country"],axis=1)

In [39]:
df_join

movie                         filename  \
0                       Seventh Son                      Seventh_Son   
1                     Welcome to Me                    Welcome_to_Me   
2                         The Judge                        The_Judge   
3   Transformers  Age of Extinction  Transformers__Age_of_Extinction   
4                  The Normal Heart                 The_Normal_Heart   
..                              ...                              ...   
90                  Live Nude Girls                  Live_Nude_Girls   
91                     Pulp Fiction                     Pulp_Fiction   
92                            Perez                            Perez   
93                    A Goofy Movie                    A_Goofy_Movie   
94                       Mean Girls                       Mean_Girls   

    goodforairplane                                            trailer  year  \
0                 1  http://www.imdb.com/video/imdb/vi1803464473/?r...  2014   
1                 0  http://www.imdb.com/video/imdb/vi226471449/?re...  2014   
2                 0  http://www.imdb.com/video/imdb/vi1286123289/?r...  2014   
3                 0  http://www.imdb.com/video/imdb/vi3138759961/?r...  2014   
4                 1  http://www.imdb.com/video/imdb/vi2749541657/?r...  2014   
..              ...                                                ...   ...   
90                0        https://www.youtube.com/watch?v=RrjF-FSFrfo  1995   
91                0        https://www.youtube.com/watch?v=s7EdQ4FqbhY  1994   
92                0        https://www.youtube.com/watch?v=HsXRKgzs1Kk  2014   
93                0        https://www.youtube.com/watch?v=Ou94SEYa9jI  1995   
94                0  http://www.imdb.com/video/imdb/vi3868524825/?r...  2004   

    runtime  rated metascore imdbRating tomatoRating  ... c_Ireland  c_Italy  \
0   102 min  PG-13        30        5.5          3.8  ...         0        0   
1   105 min      R        67        6.2          6.4  ...         0        0   
2   141 min      R        48        7.5          5.6  ...         0        0   
3   165 min  PG-13        32        5.8          3.9  ...         0        0   
4   132 min  TV-MA       N/A        8.0          7.7  ...         0        0   
..      ...    ...       ...        ...          ...  ...       ...      ...   
90   95 min      R       N/A        5.4          4.4  ...         0        0   
91  154 min      R        94        8.9          9.0  ...         0        0   
92   94 min    N/A       N/A        6.5          N/A  ...         0        1   
93   78 min      G       N/A        6.8          5.6  ...         0        0   
94   97 min  PG-13        66        7.0          6.9  ...         0        0   

    c_Japan  c_Netherlands  c_NewZealand  c_SouthAfrica  c_SouthKorea  \
0         0              0             0              0             0   
1         0              0             0              0             0   
2         0              0             0              0             0   
3         0              0             0              0             0   
4         0              0             0              0             0   
..      ...            ...           ...            ...           ...   
90        0              0             0              0             0   
91        0              0             0              0             0   
92        0              0             0              0             0   
93        0              0             0              0             0   
94        0              0             0              0             0   

    c_Spain  c_UK  c_USA  
0         0     1      1  
1         0     0      1  
2         0     0      1  
3         0     0      1  
4         0     0      1  
..      ...   ...    ...  
90        0     0      1  
91        0     0      1  
92        0     0      0  
93        0     0      1  
94        0     0      1  

[95 rows x 76 columns]

In [40]:
if user == True:
    df_join.replace("N/A", np.nan, inplace=True)


In [45]:
df_join.dtypes
df_join = df_join.drop(["movie", "filename","trailer"], axis=1)

In [43]:
df_join.columns
df_join = df_join.astype("category")

movie              category
filename           category
goodforairplane    category
trailer            category
year               category
                     ...   
c_SouthAfrica      category
c_SouthKorea       category
c_Spain            category
c_UK               category
c_USA              category
Length: 76, dtype: object

In [46]:
df_join.to_csv("meta_plus_ratings.csv",index=False)

=== Detailed Accuracy By Class === user ratings

                 TP Rate  FP Rate  Precision  Recall   F-Measure  MCC      ROC Area  PRC Area  Class
                 0,233    0,346    0,357      0,233    0,282      -0,124   0,413     0,400     '(-inf-0.5]'
                 0,654    0,767    0,507      0,654    0,571      -0,124   0,416     0,489     '(0.5-inf)'
Weighted Avg.    0,463    0,577    0,439      0,463    0,440      -0,124   0,415     0,449     



=== Detailed Accuracy By Class === meta+rating

                 TP Rate  FP Rate  Precision  Recall   F-Measure  MCC      ROC Area  PRC Area  Class
                 0,116    0,077    0,556      0,116    0,192      0,067    0,522     0,489     '(-inf-0.5]'
                 0,923    0,884    0,558      0,923    0,696      0,067    0,522     0,558     '(0.5-inf)'
Weighted Avg.    0,558    0,519    0,557      0,558    0,468      0,067    0,522     0,526     


In [107]:
## VISUAL 
# define the path
currentDirectory = pathlib.Path('D:/TU/ED/EX2/CoE_dataset/Dev_Set/vis_descriptors')

# define the pattern
currentPattern = "*.csv"
all_visual_files = []
for currentFile in currentDirectory.glob(currentPattern):
    all_visual_files.append(currentFile.parts[-1])
    
print(all_visual_files)

['American_Gangster.csv', 'American_Pie.csv', 'Andaz_Apna_Apna.csv', 'Anna_Karenina.csv', 'A_Fish_Called_Wanda.csv', 'A_Goofy_Movie.csv', 'A_Million_Ways_to_Die_in_the_West.csv', 'A_Single_Man.csv', 'Babar__The_Movie.csv', 'Bhoothnath_Returns.csv', 'Big_Hero_6.csv', 'Black_Swan.csv', 'Bobby_Jasoos.csv', 'Boogie_Nights.csv', 'Bronson.csv', 'Bros_Before_Hos.csv', 'Brother_Bear.csv', 'Casino_Royale.csv', 'Chef.csv', 'Dirty_Dancing.csv', 'Donnie_Darko.csv', 'Drive.csv', 'Elsa___Fred.csv', 'Excuse_My_French.csv', 'Exodus__Gods_and_Kings.csv', 'Full_Metal_Jacket.csv', 'Grand_Piano.csv', 'Homies.csv', 'Hook.csv', 'Hotel_Transylvania.csv', 'Inherent_Vice.csv', 'Interstellar.csv', 'Into_the_Wild.csv', 'Into_the_Woods.csv', 'Invincible.csv', 'Jack_Reacher.csv', 'Jerry_Maguire.csv', 'John_Wick.csv', 'Khoobsurat.csv', 'Khumba.csv', "Let's_Be_Cops.csv", 'Le_Samourai.csv', 'Live_Nude_Girls.csv', 'Looney_Tunes__Back_in_Action.csv', 'Lord_of_the_Flies.csv', 'Maine_Pyar_Kiya.csv', 'Mean_Girls.csv', 'Mo

In [81]:
df = pd.read_csv("vis_descriptors/" + all_visual_files[0], header=None)
df

0        1        2        3        4        5        6       7    \
0  0.23164  0.28629  0.30068  0.28118  0.24489  0.27849  0.29076  0.2938   
1  0.23164  0.28629  0.30068  0.28118  0.24489  0.27849  0.29076  0.2938   

       8        9    ...      816     817     818     819     820     821  \
0  0.16104  0.16685  ...   232.17  109.62  17.237  15.543  13.671  15.231   
1  0.16104  0.16685  ...   232.17  109.62  17.237  15.543  13.671  15.231   

     822     823    824   825  
0  53559  8637.4  18597  8679  
1  53559  8637.4  18597  8679  

[2 rows x 826 columns]

In [130]:
c = 1

for i in all_visual_files:
    if c==1:
        df_vis = pd.read_csv("vis_descriptors/" + i, header=None)
        all_features = []
        for index, row in df_vis.iterrows():
            for j in row.values:
                all_features.append(j)
        df_vis_concat = pd.DataFrame([all_features], columns=list(range(0, 1652)))
        name = i[:-4]
        df_vis_concat.insert(0, "filename", name)
        first_row = df_vis_concat.values.tolist()
        c=43
    else:
        df_vis = pd.read_csv("vis_descriptors/" + i, header=None)
        all_features = []
        for index, row in df_vis.iterrows():
            for j in row.values:
                all_features.append(j)
        df_temp = pd.DataFrame([all_features], columns=list(range(0, 1652)))
        name = i[:-4]
        df_temp.insert(0, "filename", name)
        first_row.append(df_temp.values.tolist()[0])
df_vis_fin = pd.DataFrame(first_row)
df_vis_fin

0         1         2         3     \
0                     American_Gangster  0.231640  0.286290  0.300680   
1                          American_Pie  0.237360  0.250190  0.255130   
2                       Andaz_Apna_Apna  0.000000  0.294160  0.290070   
3                         Anna_Karenina  0.000000  0.000000  0.000000   
4                   A_Fish_Called_Wanda  0.430310  0.381010  0.340820   
5                         A_Goofy_Movie  0.002031  0.000000  0.000000   
6     A_Million_Ways_to_Die_in_the_West  0.000000  0.000000  0.000000   
7                          A_Single_Man  0.000000  0.000000  0.000000   
8                      Babar__The_Movie  0.000000  0.000000  0.000000   
9                    Bhoothnath_Returns  0.229340  0.136600  0.156360   
10                           Big_Hero_6  0.271860  0.323310  0.326840   
11                           Black_Swan  0.106600  0.252310  0.372330   
12                         Bobby_Jasoos  0.032954  0.032883  0.036124   
13                        Boogie_Nights  0.136680  0.235120  0.238660   
14                              Bronson  0.000000  0.000000  0.000000   
15                      Bros_Before_Hos  0.210220  0.218160  0.325250   
16                         Brother_Bear  0.000000  0.000000  0.000000   
17                        Casino_Royale  0.000000  0.000000  0.000000   
18                                 Chef  0.281010  0.335810  0.310140   
19                        Dirty_Dancing  0.000000  0.000000  0.000000   
20                         Donnie_Darko  0.000000  0.000000  0.000000   
21                                Drive  0.000000  0.000000  0.000000   
22                          Elsa___Fred  0.310270  0.341830  0.331730   
23                     Excuse_My_French  0.000875  0.085601  0.090561   
24               Exodus__Gods_and_Kings  0.000000  0.000000  0.000000   
25                    Full_Metal_Jacket  0.179100  0.167630  0.092571   
26                          Grand_Piano  0.000000  0.000000  0.000000   
27                               Homies  0.045203  0.000000  0.000000   
28                                 Hook  0.000000  0.000000  0.000000   
29                   Hotel_Transylvania  0.000000  0.000000  0.000000   
..                                  ...       ...       ...       ...   
65                         The_Birdcage  0.000000  0.000000  0.000000   
66              The_Blair_Witch_Project  0.000000  0.000000  0.000000   
67                     The_Book_of_Life  0.000000  0.000000  0.000000   
68                    The_Bourne_Legacy  0.272070  0.295970  0.313080   
69                          The_Cobbler  0.000000  0.000000  0.000000   
70                      The_Dark_Knight  0.000000  0.000000  0.000000   
71               The_Fault_in_Our_Stars  0.296890  0.338300  0.335160   
72                     The_Great_Gatsby  0.000000  0.000000  0.000000   
73  The_Hobbit__The_Desolation_of_Smaug  0.000000  0.000000  0.000000   
74                     The_Hunger_Games  0.000000  0.000000  0.000000   
75      The_Hunger_Games__Catching_Fire  0.216510  0.286540  0.280450   
76                   The_Imitation_Game  0.236630  0.260090  0.292130   
77                            The_Judge  0.000000  0.000000  0.000000   
78                The_Last_Days_on_Mars  0.000000  0.000000  0.000651   
79             The_Lizzie_McGuire_Movie  0.000000  0.000000  0.000000   
80                           The_Matrix  0.000000  0.000000  0.000000   
81                      The_Maze_Runner  0.000000  0.000000  0.000000   
82                     The_Normal_Heart  0.000000  0.000000  0.000000   
83                      The_Parent_Trap  0.000000  0.000000  0.000000   
84                The_Phantom_Tollbooth  0.016911  0.014269  0.016807   
85                           The_Reader  0.000000  0.000000  0.000000   
86                 The_Reluctant_Dragon  0.000000  0.000000  0.000000   
87             The_Theory_of_Everything  0.000000  0.000000  0.000000   
88            This_I

In [143]:

z = list(range(0,1652))
z.insert(0, "filename")
colnames = z
df_vis_fin.columns = colnames
df_vis_fin

filename         0         1         2  \
0                     American_Gangster  0.231640  0.286290  0.300680   
1                          American_Pie  0.237360  0.250190  0.255130   
2                       Andaz_Apna_Apna  0.000000  0.294160  0.290070   
3                         Anna_Karenina  0.000000  0.000000  0.000000   
4                   A_Fish_Called_Wanda  0.430310  0.381010  0.340820   
5                         A_Goofy_Movie  0.002031  0.000000  0.000000   
6     A_Million_Ways_to_Die_in_the_West  0.000000  0.000000  0.000000   
7                          A_Single_Man  0.000000  0.000000  0.000000   
8                      Babar__The_Movie  0.000000  0.000000  0.000000   
9                    Bhoothnath_Returns  0.229340  0.136600  0.156360   
10                           Big_Hero_6  0.271860  0.323310  0.326840   
11                           Black_Swan  0.106600  0.252310  0.372330   
12                         Bobby_Jasoos  0.032954  0.032883  0.036124   
13                        Boogie_Nights  0.136680  0.235120  0.238660   
14                              Bronson  0.000000  0.000000  0.000000   
15                      Bros_Before_Hos  0.210220  0.218160  0.325250   
16                         Brother_Bear  0.000000  0.000000  0.000000   
17                        Casino_Royale  0.000000  0.000000  0.000000   
18                                 Chef  0.281010  0.335810  0.310140   
19                        Dirty_Dancing  0.000000  0.000000  0.000000   
20                         Donnie_Darko  0.000000  0.000000  0.000000   
21                                Drive  0.000000  0.000000  0.000000   
22                          Elsa___Fred  0.310270  0.341830  0.331730   
23                     Excuse_My_French  0.000875  0.085601  0.090561   
24               Exodus__Gods_and_Kings  0.000000  0.000000  0.000000   
25                    Full_Metal_Jacket  0.179100  0.167630  0.092571   
26                          Grand_Piano  0.000000  0.000000  0.000000   
27                               Homies  0.045203  0.000000  0.000000   
28                                 Hook  0.000000  0.000000  0.000000   
29                   Hotel_Transylvania  0.000000  0.000000  0.000000   
..                                  ...       ...       ...       ...   
65                         The_Birdcage  0.000000  0.000000  0.000000   
66              The_Blair_Witch_Project  0.000000  0.000000  0.000000   
67                     The_Book_of_Life  0.000000  0.000000  0.000000   
68                    The_Bourne_Legacy  0.272070  0.295970  0.313080   
69                          The_Cobbler  0.000000  0.000000  0.000000   
70                      The_Dark_Knight  0.000000  0.000000  0.000000   
71               The_Fault_in_Our_Stars  0.296890  0.338300  0.335160   
72                     The_Great_Gatsby  0.000000  0.000000  0.000000   
73  The_Hobbit__The_Desolation_of_Smaug  0.000000  0.000000  0.000000   
74                     The_Hunger_Games  0.000000  0.000000  0.000000   
75      The_Hunger_Games__Catching_Fire  0.216510  0.286540  0.280450   
76                   The_Imitation_Game  0.236630  0.260090  0.292130   
77                            The_Judge  0.000000  0.000000  0.000000   
78                The_Last_Days_on_Mars  0.000000  0.000000  0.000651   
79             The_Lizzie_McGuire_Movie  0.000000  0.000000  0.000000   
80                           The_Matrix  0.000000  0.000000  0.000000   
81                      The_Maze_Runner  0.000000  0.000000  0.000000   
82                     The_Normal_Heart  0.000000  0.000000  0.000000   
83                      The_Parent_Trap  0.000000  0.000000  0.000000   
84                The_Phantom_Tollbooth  0.016911  0.014269  0.016807   
85                           The_Reader  0.000000  0.000000  0.000000   
86                 The_Reluctant_Dragon  0.000000  0.000000  0.000000   
87             The_Theory_of_Everything  0.000000  0.000000  0.000000   
88            Th

In [182]:
df_join1 = pd.merge(df_ground, df_vis_fin, on="filename", how="inner")
df_join1 = df_join1.drop(["movie", "trailer"], axis=1)
df_join1

df_join = pd.merge(df_join, df_join1, on="filename", how="inner")
df_join  = df_join.drop(["filename", "movie", "trailer"], axis=1)
df_join


goodforairplane_x  year  runtime      rated  g_Action  g_Adventure  \
0                   1  2014  102 min      PG-13         1            1   
1                   0  2014  105 min          R         0            0   
2                   0  2014  141 min          R         0            0   
3                   0  2014  165 min      PG-13         1            1   
4                   1  2014  132 min      TV-MA         0            0   
5                   1  1970   90 min          G         0            1   
6                   1  1994  160 min         PG         0            0   
7                   1  2012   91 min         PG         0            0   
8                   1  1999  136 min          R         1            0   
9                   1  2007  148 min          R         0            1   
10                  0  1996  139 min          R         0            0   
11                  0  1990   90 min          R         0            1   
12                  0  2013  143 min      PG-13         0            0   
13                  0  2014  148 min          R         0            0   
14                  1  2001  113 min          R         0            0   
15                  1  2012  129 min          R         0            0   
16                  1  2007  157 min          R         0            0   
17                  1  1987  116 min          R         0            0   
18                  1  2013  111 min      PG-13         0            0   
19                  1  2014  101 min          R         1            0   
20                  0  2014  126 min      PG-13         0            0   
21                  0  2014   99 min      PG-13         0            0   
22                  1  2013  161 min      PG-13         0            1   
23                  0  2012  130 min      PG-13         1            0   
24                  1  2014   95 min         PG         0            1   
25                  1  2014  104 min          R         0            0   
26                  1  2009   99 min          R         0            0   
27                  1  2006  144 min      PG-13         1            1   
28                  1  2014  126 min        N/A         0            0   
29                  1  1987  100 min      PG-13         0            0   
..                ...   ...      ...        ...       ...          ...   
65                  1  1959  120 min  NOT RATED         0            0   
66                  1  2008  124 min          R         0            0   
67                  1  1996  117 min          R         0            0   
68                  1  2014  123 min      PG-13         0            0   
69                  1  2010  108 min          R         0            0   
70                  0  1989  192 min        N/A         0            0   
71                  0  2014  101 min      PG-13         0            0   
72                  0  2011   99 min          R         0            0   
73                  0  2014  155 min        N/A         0            0   
74                  1  2013   90 min          R         0            0   
75                  1  2014  125 min         PG         0            1   
76                  1  1999   81 min          R         0            0   
77                  1  2013  104 min          G         0            1   
78                  1  2013  132 min        N/A         0            0   
79                  1  2014  103 min          R         0            0   
80                  0  2012  135 min      PG-13         1            1   
81                  0  2014  122 min          R         0            0   
82                  0  2008  152 min      PG-13         1            0   
83                  0  2014   97 min          R         0            0   
84                  1  2010  140 min      PG-13         1            1   
85                  1  2014  116 min          R         0            0   
86                  0  2011  100 min          R         0            0   
87      

In [183]:
df_join.to_csv("meta+visual.csv",index=False)

=== Detailed Accuracy By Class === visual

                 TP Rate  FP Rate  Precision  Recall   F-Measure  MCC      ROC Area  PRC Area  Class
                 0,163    0,154    0,467      0,163    0,241      0,012    0,485     0,445     '(-inf-0.5]'
                 0,846    0,837    0,550      0,846    0,667      0,012    0,485     0,543     '(0.5-inf)'
Weighted Avg.    0,537    0,528    0,512      0,537    0,474      0,012    0,485     0,499 


=== Detailed Accuracy By Class === visual+meta

                 TP Rate  FP Rate  Precision  Recall   F-Measure  MCC      ROC Area  PRC Area  Class
                 0,233    0,231    0,455      0,233    0,308      0,002    0,482     0,442     '(-inf-0.5]'
                 0,769    0,767    0,548      0,769    0,640      0,002    0,482     0,536     '(0.5-inf)'
Weighted Avg.    0,526    0,525    0,506      0,526    0,490      0,002    0,482     0,493 